widget/list.go

```diff
@@ -250,23 +251,39 @@ func (l *List) UnselectAll() {
 	}
 }
 
-func (l *List) visibleItemHeights(itemHeight float32, length int) (visible map[int]float32, offY float32, minRow, maxRow int) {
+func (l *List) visibleItemHeights(itemHeight float32, length int) (visible []float32, offY float32, minRow, maxRow int) {
 	maxRow = length
 	rowOffset := float32(0)
 	isVisible := false
-	visible = make(map[int]float32)
+	visible = []float32{}
 
 	if l.scroller.Size().Height <= 0 {
 		return
 	}
 
+	// theme.Padding is a slow call, so we cache it
+	padding := theme.Padding()
+
+	if len(l.itemHeights) == 0 {
+		offY = l.offsetY - itemHeight - padding
```

Comment:
This doesn't seem right - it should be the offset of the first visible item which is aligned on row boundaries.
This calculation looks like it is always 1 cell off-screen... what about when at 0, or when an item is half visible?

Reply:
More likely the fast way to calculate would be offsetY modulus padded height.